src/python/pants/backend/plugin_development/goals/check_names.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""A Plugin development tool, to help check that naming conventions are followed in all loaded
+backends.
+
+Usage:
+
+    $ ./pants --backend-packages=pants.backend.plugin_development check-names
+
+Implemented checks:
+
+  * Field classes should have a `Field` suffix on the class name. (`Mixin` and `Base` are also
+    accepted, but not advertised.)
+"""
+
+from __future__ import annotations
+
+from typing import Iterator
+
+from pants.engine.console import Console
+from pants.engine.goal import Goal, GoalSubsystem
+from pants.engine.rules import collect_rules, goal_rule
+from pants.engine.target import Field
+
+
+class CheckNamesSubsystem(GoalSubsystem):
+    name = "check-names"
+    help = "Check names of things, that they follow Pants naming standards."
+
+
+class CheckNames(Goal):
+    subsystem_cls = CheckNamesSubsystem
+
+
+@goal_rule
+async def check_names(console: Console, subsystem: CheckNamesSubsystem) -> CheckNames:
+    console.print_stdout("Checking field class names...")
+    exit_code = check_field_names(console)
+
+    if exit_code == 0:
+        sigil = console.sigil_succeeded()
+        status = "all checks passed"
+    else:
+        sigil = console.sigil_failed()
+        status = "some checks failed"
+
+    console.print_stdout(f"\n{sigil} {status}.")
+    return CheckNames(exit_code)
+
+
+def get_subclasses_name_does_not_end_with(
+    base_class: type, *endswith: str
+) -> Iterator[tuple[str, str]]:
+    for cls in base_class.__subclasses__():
```

Comment:
Woah, fancy

Reply:
Yeah, I've found the `cls.__subclasses__()` real handy, it saves on registration boilerplate a lot, when you want a registry of classes belonging to a certain category..